### PR TITLE
Text

### PR DESCRIPTION
**Pull request by OroArmor**

### CODE REVIEW

File: mappings/net/minecraft/text/MutableText.mapping [hunk 1]
```diff
@@ -1,5 +1,16 @@
 CLASS net/minecraft/unmapped/C_npqneive net/minecraft/text/MutableText
 	COMMENT A text with mutation operations.
+	FIELD f_fggxfgch language Lnet/minecraft/unmapped/C_gedhbjjb;
+	FIELD f_fhizxfyx ordered Lnet/minecraft/unmapped/C_apvkgwyi;
+	FIELD f_gwqwjofn style Lnet/minecraft/unmapped/C_cpwnhism;
+	FIELD f_rvaoapfk component Lnet/minecraft/unmapped/C_hhxmszuy;
+	FIELD f_ulczlyna siblings Ljava/util/List;
+	METHOD <init> (Lnet/minecraft/unmapped/C_hhxmszuy;Ljava/util/List;Lnet/minecraft/unmapped/C_cpwnhism;)V
+		ARG 1 component
+		ARG 2 siblibgs
```
Comment (triphora) [inline mappings/net/minecraft/text/MutableText.mapping]:
```suggestion
		ARG 2 siblings
```

File: mappings/net/minecraft/text/Style.mapping [hunk 2]
```diff
@@ -143,6 +143,15 @@ CLASS net/minecraft/unmapped/C_cpwnhism net/minecraft/text/Style
 		COMMENT
 		COMMENT <p>An insertion is inserted when a piece of text clicked while shift key
 		COMMENT is down in the chat HUD.
+	CLASS C_azplftfl Builder
+		FIELD f_szmdllpx hasAddedElement Z
+		METHOD m_ndzksqsj addSeperator ()V
```
Comment (triphora) [inline mappings/net/minecraft/text/Style.mapping]:
```suggestion
		METHOD m_ndzksqsj addSeparator ()V
```

File: mappings/net/minecraft/text/component/SelectorComponent.mapping [hunk 3]
```diff
@@ -0,0 +1,15 @@
+CLASS net/minecraft/unmapped/C_mvxjmafd net/minecraft/text/component/SelectorComponent
+	FIELD f_liyiswnc seperator Ljava/util/Optional;
```
Comment (triphora) [inline mappings/net/minecraft/text/component/SelectorComponent.mapping]:
```suggestion
	FIELD f_liyiswnc separator Ljava/util/Optional;
```

File: mappings/net/minecraft/text/component/SelectorComponent.mapping [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+	FIELD f_qowittym LOGGER Lorg/slf4j/Logger;
+	METHOD <init> (Ljava/lang/String;Ljava/util/Optional;)V
+		ARG 1 string
+		ARG 2 seperator
+	METHOD equals (Ljava/lang/Object;)Z
+		ARG 1 o
+	METHOD m_dqzswnww getSelector (Ljava/lang/String;)Lnet/minecraft/unmapped/C_nbqdyigm;
+		ARG 0 string
+	METHOD m_iuqfhipv getEntitySelector ()Lnet/minecraft/unmapped/C_nbqdyigm;
+	METHOD m_ojiagyez getSeperator ()Ljava/util/Optional;
```
Comment (triphora) [inline mappings/net/minecraft/text/component/SelectorComponent.mapping]:
```suggestion
	METHOD m_ojiagyez getSeparator ()Ljava/util/Optional;
```

File: mappings/net/minecraft/text/component/SelectorComponent.mapping [hunk 5]
```diff
@@ -0,0 +1,15 @@
+CLASS net/minecraft/unmapped/C_mvxjmafd net/minecraft/text/component/SelectorComponent
+	FIELD f_liyiswnc seperator Ljava/util/Optional;
+	FIELD f_npzvycba string Ljava/lang/String;
+	FIELD f_paqwsnhp entitySelector Lnet/minecraft/unmapped/C_nbqdyigm;
+	FIELD f_qowittym LOGGER Lorg/slf4j/Logger;
+	METHOD <init> (Ljava/lang/String;Ljava/util/Optional;)V
+		ARG 1 string
+		ARG 2 seperator
```
Comment (triphora) [inline mappings/net/minecraft/text/component/SelectorComponent.mapping]:
```suggestion
		ARG 2 separator
```

File: mappings/net/minecraft/text/component/NbtComponent.mapping [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+		ARG 1 textLeft
+		ARG 2 textRight
+	METHOD m_caghjkep isInterpreting ()Z
+	METHOD m_eihlegen (Lnet/minecraft/unmapped/C_rdaqiwdt;Lnet/minecraft/unmapped/C_npqneive;Lnet/minecraft/unmapped/C_npqneive;)Lnet/minecraft/unmapped/C_npqneive;
+		ARG 1 textLeft
+		ARG 2 textRight
+	METHOD m_fyflhfse getPathPattern ()Ljava/lang/String;
+	METHOD m_glyxacly getSeperator ()Ljava/util/Optional;
+	METHOD m_lirddokf (Lnet/minecraft/unmapped/C_hhlwcnih;)Ljava/util/stream/Stream;
+		ARG 1 compond
```
Comment (triphora) [inline mappings/net/minecraft/text/component/NbtComponent.mapping]:
```suggestion
		ARG 1 compound
```

File: mappings/net/minecraft/text/component/NbtComponent.mapping [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+		ARG 1 o
+	METHOD m_bklvalmj (Lnet/minecraft/unmapped/C_npqneive;Lnet/minecraft/unmapped/C_npqneive;Lnet/minecraft/unmapped/C_npqneive;)Lnet/minecraft/unmapped/C_npqneive;
+		ARG 1 textLeft
+		ARG 2 textRight
+	METHOD m_caghjkep isInterpreting ()Z
+	METHOD m_eihlegen (Lnet/minecraft/unmapped/C_rdaqiwdt;Lnet/minecraft/unmapped/C_npqneive;Lnet/minecraft/unmapped/C_npqneive;)Lnet/minecraft/unmapped/C_npqneive;
+		ARG 1 textLeft
+		ARG 2 textRight
+	METHOD m_fyflhfse getPathPattern ()Ljava/lang/String;
+	METHOD m_glyxacly getSeperator ()Ljava/util/Optional;
```
Comment (triphora) [inline mappings/net/minecraft/text/component/NbtComponent.mapping]:
```suggestion
	METHOD m_glyxacly getSeparator ()Ljava/util/Optional;
```

File: mappings/net/minecraft/text/component/NbtComponent.mapping [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+	METHOD <init> (Ljava/lang/String;Lnet/minecraft/unmapped/C_xondkbcs$C_rgpfrkey;ZLjava/util/Optional;Lnet/minecraft/unmapped/C_pfocmsrt;)V
+		ARG 1 pathPattern
+		ARG 2 path
+		ARG 3 interpreting
+		ARG 4 seperator
+		ARG 5 nbt
+	METHOD <init> (Ljava/lang/String;ZLjava/util/Optional;Lnet/minecraft/unmapped/C_pfocmsrt;)V
+		ARG 1 pathPattern
+		ARG 2 interpreting
+		ARG 3 seperator
```
Comment (triphora) [inline mappings/net/minecraft/text/component/NbtComponent.mapping]:
```suggestion
		ARG 3 separator
```

File: mappings/net/minecraft/text/component/NbtComponent.mapping [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+	FIELD f_dxprtbyj interpreting Z
+	FIELD f_ffcigvcs seperator Ljava/util/Optional;
+	FIELD f_imwirfmg path Lnet/minecraft/unmapped/C_xondkbcs$C_rgpfrkey;
+	FIELD f_inmzzmlo LOGGER Lorg/slf4j/Logger;
+	FIELD f_suokeamh pathPattern Ljava/lang/String;
+	METHOD <init> (Ljava/lang/String;Lnet/minecraft/unmapped/C_xondkbcs$C_rgpfrkey;ZLjava/util/Optional;Lnet/minecraft/unmapped/C_pfocmsrt;)V
+		ARG 1 pathPattern
+		ARG 2 path
+		ARG 3 interpreting
+		ARG 4 seperator
```
Comment (triphora) [inline mappings/net/minecraft/text/component/NbtComponent.mapping]:
```suggestion
		ARG 4 separator
```

File: mappings/net/minecraft/text/component/NbtComponent.mapping [hunk 10]
```diff
@@ -0,0 +1,38 @@
+CLASS net/minecraft/unmapped/C_tvzpgqlc net/minecraft/text/component/NbtComponent
+	FIELD f_dmcxfrec nbt Lnet/minecraft/unmapped/C_pfocmsrt;
+	FIELD f_dxprtbyj interpreting Z
+	FIELD f_ffcigvcs seperator Ljava/util/Optional;
```
Comment (triphora) [inline mappings/net/minecraft/text/component/NbtComponent.mapping]:
```suggestion
	FIELD f_ffcigvcs separator Ljava/util/Optional;
```

**Review by triphora — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/text/Text.mapping [hunk 11]
```diff
@@ -12,13 +12,24 @@ CLASS net/minecraft/unmapped/C_rdaqiwdt net/minecraft/text/Text
...
+	METHOD m_nxgujqbg createKeybind (Ljava/lang/String;)Lnet/minecraft/unmapped/C_npqneive;
+		ARG 0 text
+	METHOD m_oamaavje empty ()Lnet/minecraft/unmapped/C_npqneive;
 	METHOD m_oyjmhrxe getStyle ()Lnet/minecraft/unmapped/C_cpwnhism;
 		COMMENT Returns the style of this text.
 	METHOD m_pnqfqdco getWithStyle (Lnet/minecraft/unmapped/C_cpwnhism;)Ljava/util/List;
 		ARG 1 style
-	METHOD m_pwvegngp asString ()Lnet/minecraft/unmapped/C_hhxmszuy;
+	METHOD m_pwvegngp asComponent ()Lnet/minecraft/unmapped/C_hhxmszuy;
 		COMMENT Returns the string representation of this text itself, excluding siblings.
```
Comment (IotaBread) [inline mappings/net/minecraft/text/Text.mapping]: This javadoc needs updating

File: mappings/net/minecraft/text/Text.mapping [hunk 12]
```diff
@@ -32,6 +43,16 @@ CLASS net/minecraft/unmapped/C_rdaqiwdt net/minecraft/text/Text
 	METHOD m_ucpxjqgq (Ljava/util/List;Lnet/minecraft/unmapped/C_cpwnhism;Ljava/lang/String;)Ljava/util/Optional;
 		ARG 1 styleOverride
 		ARG 2 text
+	METHOD m_uysfrnzl create (Ljava/lang/String;)Lnet/minecraft/unmapped/C_npqneive;
+		ARG 0 text
+	METHOD m_vwzgcgyh createNbt (Ljava/lang/String;ZLjava/util/Optional;Lnet/minecraft/unmapped/C_pfocmsrt;)Lnet/minecraft/unmapped/C_npqneive;
+		ARG 0 text
+		ARG 1 interpreting
+		ARG 2 seperater
```
Comment (IotaBread) [inline mappings/net/minecraft/text/Text.mapping]:
```suggestion
		ARG 2 separator
```

File: mappings/net/minecraft/text/component/KeybindComponent.mapping [hunk 13]
```diff
@@ -0,0 +1,9 @@
+CLASS net/minecraft/unmapped/C_scqpkchq net/minecraft/text/component/KeybindComponent
```
Comment (IotaBread) [inline mappings/net/minecraft/text/component/KeybindComponent.mapping:1]: Don't we use text instead of component?

Comment (OroArmor) [reply mappings/net/minecraft/text/component/KeybindComponent.mapping:1]: These are objects that turn into `Mutable` text objects

File: mappings/net/minecraft/text/Text.mapping (hunk 11 above)
Comment (OroArmor) [reply mappings/net/minecraft/text/Text.mapping]:
```suggestion
		COMMENT Returns a component representation of this text.
```

File: mappings/net/minecraft/text/Text.mapping [hunk 14]
```diff
@@ -12,13 +12,24 @@ CLASS net/minecraft/unmapped/C_rdaqiwdt net/minecraft/text/Text
...
+	METHOD m_dztmuipa createFormatted (Ljava/lang/String;[Ljava/lang/Object;)Lnet/minecraft/unmapped/C_npqneive;
+		ARG 0 text
+		ARG 1 args
 	METHOD m_eiqxivnk getSiblings ()Ljava/util/List;
 		COMMENT Returns the siblings of this text.
+	METHOD m_iktyeyef createFormatted (Ljava/lang/String;)Lnet/minecraft/unmapped/C_npqneive;
+		ARG 0 text
+	METHOD m_kuqyhsxe createSelector (Ljava/lang/String;Ljava/util/Optional;)Lnet/minecraft/unmapped/C_npqneive;
+		ARG 0 name
+		ARG 1 seperator
```
Comment (triphora) [inline mappings/net/minecraft/text/Text.mapping]:
```suggestion
		ARG 1 separator
```

**Review by triphora — APPROVED** [review]

**Review by IotaBread — APPROVED** [review]

**Pull request merged**